### PR TITLE
feat(pkg/kube): add openapi validation for k8s objects

### PR DESCRIPTION
**Pull request by adamreese**
Add back OpenAPI validation for kubernetes objects.

Fixes: #6382

### CODE REVIEW

**Review by bacongobbler — APPROVED** [review]

**Comment by bacongobbler** [comment]
Seems fairly straightforward. LGTM!

File: pkg/kube/client.go [hunk 1]
```diff
@@ -122,10 +122,15 @@ func (c *Client) newBuilder() *resource.Builder {
...
-func (c *Client) Build(reader io.Reader) (ResourceList, error) {
+func (c *Client) Build(reader io.Reader, validate bool) (ResourceList, error) {
+	schema, err := c.Factory.Validator(validate)
+	if err != nil {
+		return nil, err
+	}
 	result, err := c.newBuilder().
 		Unstructured().
 		Stream(reader, "").
+		Schema(schema).
```
Comment (hickeyma) [inline pkg/kube/client.go]: Should `Schema` be before `Stream` as per https://github.com/helm/helm/pull/5576 ?

**Review by hickeyma — CHANGES_REQUESTED** [review]
@adamreese Thanks for this. Looks good but could you take a  look at the order in builder?

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @adamreese

**Pull request merged**